pkg/util/transport_util.go

```diff
@@ -78,5 +94,20 @@ func MakeTransport(opts config.RegistryOptions, registryName string) http.RoundT
 			}
 		}
 	}
+
+	if clientCertificatePath := opts.RegistriesClientCertificates[registryName]; clientCertificatePath != "" {
+		certFiles := strings.Split(clientCertificatePath, ",")
+		if len(certFiles) != 2 {
+			logrus.Warnf("Failed to load client certificate/key '%s=%s', expected format: %s=/path/to/cert,/path/to/key\n", registryName, clientCertificatePath, registryName)
```

Comment:
I think this should be an error, rather than logging and otherwise ignoring the invalid cert/key pair.

Reply:
Hi @imjasonh, I was trying to match the style/behavior of the certificate authority flag (`--registry-certificate`) when I made this a warning.

After writing this and looking at it now, I agree with you that this should be a fatal error; someone said "please use these certs" and it didn't find them.

Just to confirm, you were requesting this to be a fatal error and exit with an error code if the certificate is not loaded?

Reply:
Yes that's correct. If the flag is malformed (e.g., `registry.com=/path/to/cert,/path/to/key,/some/other/path`) or if the correctly specified file(s) can't be loaded, these should be fatal errors.

Same with registry certs above, now that you mention it, for consistency.

It's likely (hopefully) that not being able to load the certs will cause executions to fail later anyway, but it'd be better to fail as soon as we tell something is wrong here, instead of letting the build proceed and fail later (again, hopefully) if the necessary certs aren't available.

Reply:
Hi @imjasonh,

I think this looks a lot better now, since the error is stated immediately. I think it's a lot easier to identify potential issues.

Here is the before/after (with a simple typo of omitting the `m` in `pem`):

## Before

```
docker run ... kaniko-debug ... --registry-certificate test.example.com=/certs/ca.pe
WARN[0000] Failed to load certificate /certs/ca.pe for test.example.com
  error="open /certs/ca.pe: no such file or directory"
error checking push permissions -- make sure you entered the correct tag name,
  and that you are authenticated correctly, and try again:
  checking push permission for "test.example.com/example/example:latest":
  creating push check transport for test.example.com failed:
  Get "https://test.example.com/v2/": x509: certificate signed by unknown authority
```

## After this PR

```
docker run ... kaniko-debug ... --registry-certificate test.example.com=/certs/ca.pe
error checking push permissions -- make sure you entered the correct tag name,
  and that you are authenticated correctly, and try again:
  making transport for registry "test.example.com":
  failed to load certificate /certs/ca.pe for test.example.com:
  open /certs/ca.pe: no such file or directory
```

I did resolve this for both the verification (authority) and client certs.

WDYT?